File: src/main/java/com/extension/factcheck/application/UserService.java
package com.extension.factcheck.application;

import com.extension.factcheck.domain.User;
import com.extension.factcheck.domain.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@RequiredArgsConstructor
@Service
public class UserService {

  private final UserRepository userRepository;

  @Transactional(readOnly = true)
  public User findById(Long id) {
    return userRepository.findById(id)
        .orElseThrow(() -> new IllegalArgumentException("User not found with id: " + id));
  }
}

File: src/main/kotlin/com/extension/factcheck/application/UserService.kt
package com.extension.factcheck.application

import com.extension.factcheck.domain.User
import com.extension.factcheck.domain.UserRepository
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class UserService(
    private val userRepository: UserRepository,
) {

    @Transactional(readOnly = true)
    fun findById(id: Long): User {
        return userRepository.findByIdOrNull(id)
            ?: throw IllegalArgumentException("User with ID $id not found")
    }
}
